spec/lib/alaveteli_geoip_spec.rb

```diff
@@ -0,0 +1,123 @@
+# -*- encoding : utf-8 -*-
+require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
+
+describe AlaveteliGeoIP do
+
+	describe '.country_code_from_ip' do
+
+    it 'delegates to an instance of the class' do
+      instance = double
+      allow(AlaveteliGeoIP).to receive(:instance).and_return(instance)
+      allow(instance).to receive(:country_code_from_ip).with('127.0.0.1')
+      AlaveteliGeoIP.country_code_from_ip '127.0.0.1'
+    end
+
+	end
+
+  describe '.instance' do
+
+    it 'creates a new instance' do
+      expect(AlaveteliGeoIP.instance).to be_instance_of(AlaveteliGeoIP)
+    end
+
+    it 'caches the instance' do
+      expect(AlaveteliGeoIP.instance).to equal(AlaveteliGeoIP.instance)
+    end
+
+  end
+
+  describe '.new' do
+
+    it 'configures the instance with the configured country code' do
```

Comment:
Ah, sorry I got this confused with configuring the default database (https://github.com/mysociety/alaveteli/pull/2734/files#r37849670)